research/tensorrt/README.md

```diff
@@ -0,0 +1,147 @@
+# Running the TensorFlow Official ResNet with TensorRT
+
+[TensorRT](https://developer.nvidia.com/tensorrt) is NVIDIA's inference
+optimizer for deep learning. Briefly, TensorRT rewrites parts of the
+execution graph to allow for faster prediction times.
+
+Here we provide a sample script that can:
+
+1. Convert a TensorFlow SavedModel to a Frozen Graph.
+2. Load a Frozen Graph for inference.
+3. Time inference loops using the native TensorFlow graph.
+4. Time inference loops using FP32, FP16, or INT8<sup>1</sup> precision modes from TensorRT.
+
+We provide some results below, as well as instructions for running this script.
+
+<sup>1</sup> INT8 mode is a work in progress; please see [INT8 Mode is the Bleeding Edge](#int8-mode-is-the-bleeding-edge) below.
+
+## How to Run This Script
+
+### Step 1: Install Prerequisites
+
+1. [Install TensorFlow.](https://www.tensorflow.org/install/)
+2. [Install TensorRT.](http://docs.nvidia.com/deeplearning/sdk/tensorrt-install-guide/index.html)
+3. We use the [Official version of ResNet image preprocessing](/official/resnet/imagenet_preprocessing.py). Please checkout the Models repository if you haven't
+already, and add the Official Models to your Python path:
+
+```
+cd /path/to
+git clone https://github.com/tensorflow/models.git
+export PYTHONPATH="$PYTHONPATH:/path/to/models"
+```
+
+### Step 2: Get a model to test
+
+The provided script runs with the [Official version of ResNet trained with
+ImageNet data](/official/resnet), but can be used for other models as well,
+as long as you have a SavedModel or a Frozen Graph.
+
+You can download the ResNetv2-ImageNet [SavedModel](http://download.tensorflow.org/models/official/resnetv2_imagenet_savedmodel.tar.gz)
+or [Frozen Graph](http://download.tensorflow.org/models/official/resnetv2_imagenet_frozen_graph.pb),
+or, if you want to train the model yourself,
+pass `--export_dir` to the Official ResNet [imagenet_main.py](/official/resnet/imagenet_main.py).
+
+When running this script, you can pass in a SavedModel directory containing the
+Protobuf MetaGraphDef and variables directory to `savedmodel_dir`, or pass in
+a Protobuf frozen graph file directly to `frozen_graph`. If you downloaded the
+SavedModel linked above, note that you should untar it before passing in to the
+script.
+
+### Step 3: Get an image to test
+
+The script can accept a JPEG image file to use for predictions. If none is
```

Comment:
Are there any requirements for the image, other than the JPEG format?

Reply:
I don't think so, assuming that we all assume that the picture should have something identifiable and worth labeling in it.